From 0ea11ba24e38635212ca41fa697b51d7990fe1d2 Mon Sep 17 00:00:00 2001
From: Afzal Mohammed <afzal@ti.com>
Date: Thu, 12 Jan 2012 10:47:36 +0530
Subject: [PATCH 0699/1092] regulator: TPS65910: VDD1 error handling

Proper error handling w.r.t I2C transactions
for setting & getting voltage for VDD1

Signed-off-by: Afzal Mohammed <afzal@ti.com>
---
 drivers/regulator/tps65910-regulator.c | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/drivers/regulator/tps65910-regulator.c b/drivers/regulator/tps65910-regulator.c
index bb2a08d..fc42a34 100644
--- a/drivers/regulator/tps65910-regulator.c
+++ b/drivers/regulator/tps65910-regulator.c
@@ -484,9 +484,15 @@ static int tps65910_get_voltage_dcdc(struct regulator_dev *dev)
 	switch (id) {
 	case TPS65910_REG_VDD1:
 		opvsel = tps65910_reg_read(pmic, TPS65910_VDD1_OP);
+		if (opvsel < 0)
+			return opvsel;
 		mult = tps65910_reg_read(pmic, TPS65910_VDD1);
+		if (mult < 0)
+			return mult;
 		mult = (mult & VDD1_VGAIN_SEL_MASK) >> VDD1_VGAIN_SEL_SHIFT;
 		srvsel = tps65910_reg_read(pmic, TPS65910_VDD1_SR);
+		if (srvsel < 0)
+			return srvsel;
 		sr = opvsel & VDD1_OP_CMD_MASK;
 		opvsel &= VDD1_OP_SEL_MASK;
 		srvsel &= VDD1_SR_SEL_MASK;
@@ -637,6 +643,7 @@ static int tps65910_set_voltage_dcdc(struct regulator_dev *dev,
 	struct tps65910_reg *pmic = rdev_get_drvdata(dev);
 	int id = rdev_get_id(dev), vsel;
 	int dcdc_mult = 0;
+	int ret = 0;
 
 	switch (id) {
 	case TPS65910_REG_VDD1:
@@ -645,10 +652,11 @@ static int tps65910_set_voltage_dcdc(struct regulator_dev *dev,
 			dcdc_mult--;
 		vsel = (selector % VDD1_2_NUM_VOLT_FINE) + 3;
 
-		tps65910_modify_bits(pmic, TPS65910_VDD1,
+		ret = tps65910_modify_bits(pmic, TPS65910_VDD1,
 				(dcdc_mult << VDD1_VGAIN_SEL_SHIFT),
 						VDD1_VGAIN_SEL_MASK);
-		tps65910_reg_write(pmic, TPS65910_VDD1_OP, vsel);
+		if (!ret)
+			ret = tps65910_reg_write(pmic, TPS65910_VDD1_OP, vsel);
 		break;
 	case TPS65910_REG_VDD2:
 		dcdc_mult = (selector / VDD1_2_NUM_VOLT_FINE) + 1;
@@ -666,7 +674,7 @@ static int tps65910_set_voltage_dcdc(struct regulator_dev *dev,
 		tps65910_reg_write(pmic, TPS65911_VDDCTRL_OP, vsel);
 	}
 
-	return 0;
+	return ret;
 }
 
 static int tps65910_set_voltage(struct regulator_dev *dev, unsigned selector)
-- 
1.7.11.2

